lib/http_server/router.ex

```diff
@@ -1,17 +1,19 @@
-defmodule Router do
-    alias Response.Builder
-    alias Request.Parser
+defmodule HttpServer.Router do
+    alias HttpServer.Exception
+    alias HttpServer.Request.Parser
+    alias HttpServer.Response
+    alias HttpServer.Response.Builder
 
-    def handle_request(http_message, routes \\ Routes) do
+    def handle_request(http_message, routes) do
         request = parse_http_message(http_message)
 
-        handler = route_to_handler(request)
+        handler = route_to_handler(routes, request)
         
         response = handler.handle(request, Builder)
         Response.format(response)
     end
 
-    def route_to_handler(request, routes \\ Routes) do
+    def route_to_handler(routes, request) do
         routes.build_routes_map()
         |> get_method_routes(request.method) 
         |> get_handler(request.resource)
```

Comment:
If we move `Routes` out of `HttpServer.HttpServer` 

```suggestion
    def route_to_handler(request) do
        Routes.build_routes_map()
        |> get_method_routes(request.method) 
        |> get_handler(request.resource)
```

Reply:
I actually think you may have suggested this earlier, but I only see it now.

Reply:
No, I didn't. I think it's a good suggestion, but we we can't mock `Routes` for tests if we make these changes.

Reply:
ok, then maybe go back to your original: 
```suggestion
    def route_to_handler(request, routes \\ Routes) do
        routes.build_routes_map()
        |> get_method_routes(request.method) 
        |> get_handler(request.resource)
```
and make sure to:
```elixir
alias Routes
```

Reply:
do you think that would work?  all these comments are basically around getting the Routes injection out of the `HttpServer.HttpServer`.  If that doesn't make sense to you, let me know.  I think I approve of this PR either way.